### DISCUSSION TITLE
How to add users in auth in Supabase self-hosted?

### DISCUSSION THREAD

**Discussion by edwardcwang** [supabase/supabase#7912] [Questions] [answered]
In Supabase self-hosted, how do we add users? The platform version has a green "Invite" button which is missing in the self-hosted version.

Thank you for your time and attention in advance!

![4](https://user-images.githubusercontent.com/10752510/181119153-be6e3049-81d7-42d3-8331-ed5f6dcc25f9.png)

**Comment by wcooperscott** [comment, marked as answer]
I am not certain if this is the _best_ method, but it is a working method.

You can use the Supabase-JS client, authenticated with the service_role API key to create accounts. I am using this code example on an edge function, but you could run it on your local computer:
`const signup_result = await supabase.auth.api.signUpWithEmail(email, password);`

**Reply by edwardcwang** [reply]
Works great; thank you so much!

Note: requires `DISABLE_SIGNUP` to be false, even if using the `service_role` key.

**Reply by WilliamScott-AdventGX** [reply]
Glad to hear it! 👍
